samples/net/irc_bot/src/irc-bot.c

```diff
@@ -43,179 +45,144 @@ static u8_t cmd_buf[CMD_BUFFER_SIZE];
+static struct net_buf_pool *data_pool(void)
+{
+	return &data_tcp;
+}
+#else
+#define tx_slab NULL
+#define data_pool NULL
+#endif /* CONFIG_NET_CONTEXT_NET_PKT_POOL */
 
 static void
 panic(const char *msg)
 {
-	NET_ERR("Panic: %s", msg);
+	SYS_LOG_ERR("Panic: %s", msg);
 	for (;;) {
 		k_sleep(K_FOREVER);
 	}
 }
 
 static int
-transmit(struct net_context *ctx, char buffer[], size_t len)
+transmit(char buffer[], size_t len)
 {
 	struct net_pkt *send_pkt;
+	int ret;
 
-	send_pkt = net_pkt_get_tx(ctx, K_FOREVER);
+	send_pkt = net_app_get_net_pkt(&app_ctx, AF_UNSPEC, BUF_ALLOC_TIMEOUT);
 	if (!send_pkt) {
+		SYS_LOG_ERR("Unable to get TX packet, not enough memory.");
 		return -ENOMEM;
 	}
 
-	if (!net_pkt_append_all(send_pkt, len, (u8_t *)buffer, K_FOREVER)) {
+	if (!net_pkt_append_all(send_pkt, len, (u8_t *)buffer,
+				BUF_ALLOC_TIMEOUT)) {
+		net_pkt_unref(send_pkt);
 		return -EINVAL;
 	}
 
-	return net_context_send(send_pkt, NULL, K_NO_WAIT, NULL, NULL);
+	ret = net_app_send_pkt(&app_ctx, send_pkt,
+			       &app_ctx.default_ctx->remote,
+			       NET_SOCKADDR_MAX_SIZE, K_NO_WAIT, NULL);
+	if (ret < 0) {
+		net_pkt_unref(send_pkt);
+	}
+
+	return ret;
 }
 
 static void
-on_cmd_ping(struct zirc *irc, char *umask, char *cmd, size_t len)
+on_cmd_ping(char *umask, char *cmd, size_t len)
 {
 	char pong[32];
 	int ret;
 
-	NET_INFO("Got PING command from server: %s", cmd);
+	SYS_LOG_INF("Got PING command from server: %s", cmd);
 
 	ret = snprintk(pong, 32, "PONG :%s", cmd + 1);
 	if (ret < sizeof(pong)) {
-		ret = transmit(irc->conn, pong, ret);
+		ret = transmit(pong, ret);
 		if (ret < 0) {
-			NET_INFO("Transmit error: %d", ret);
+			SYS_LOG_INF("Transmit error: %d", ret);
 		}
 	}
 }
 
 static void
-on_cmd_privmsg(struct zirc *irc, char *umask, char *cmd, size_t len)
+on_cmd_privmsg(char *umask, char *cmd, size_t len)
 {
-	struct zirc_chan *chan;
 	char *space;
 
 	if (!umask) {
 		/* Don't know how this got here, so ignore */
 		return;
 	}
 
-	NET_INFO("Got message from umask %s: %s", umask, cmd);
+	SYS_LOG_DBG("Got message from umask %s: %s", umask, cmd);
 
+	/* find space between channel/username */
 	space = memchr(cmd, ' ', len);
 	if (!space) {
 		return;
 	}
-
 	*space = '\0';
 
 	if (*(space + 1) != ':') {
 		/* Just ignore messages without a ':' after the space */
+		SYS_LOG_DBG("Ignoring message umask: %s: %s", umask, space);
 		return;
 	}
 
 	space += 2; /* Jump over the ':', pointing to the message itself */
-	for (chan = irc->chans; chan; chan = chan->next) {
-		if (!strncmp(chan->chan, cmd, space - cmd)) {
-			chan->on_privmsg_rcvd(chan->data, chan, umask, space);
-			return;
-		}
+
+	/* check for a private message from another user */
+	if (!strncmp(nick_buf, cmd, NICK_BUFFER_SIZE)) {
+		on_privmsg_rcvd(umask, umask, space);
+	} else {
+		on_privmsg_rcvd(cmd, umask, space);
 	}
 
-	/* TODO: could be a private message (from another user) */
-	NET_INFO("Unknown privmsg received: %.*s\n", (int)len, cmd);
+	return;
```

Comment:
Stray `return;`?

Reply:
Not sure how that got in there.. probably the same.  Thanks!